packages/storage/amplify_storage_s3_dart/lib/src/model/storage_s3_item.dart

```diff
@@ -0,0 +1,66 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:amplify_core/amplify_core.dart';
+import 'package:amplify_storage_s3_dart/amplify_storage_s3_dart.dart';
+import 'package:amplify_storage_s3_dart/src/sdk/s3.dart';
+import 'package:meta/meta.dart';
+
+/// {@template storage.amplify_storage_s3.storage_s3_item}
+/// This defines an object in a S3 bucket.
+/// {@endtemplate}
+class StorageS3Item extends StorageItem {
+  /// {@macro storage.amplify_storage_s3.storage_s3_item}
+  StorageS3Item({
+    required super.key,
+    super.size,
+    super.lastModified,
+    super.eTag,
+    this.metadata = const <String, String>{},
+    this.versionId,
+  });
+
+  /// Creates a [StorageS3Item] from [S3Object] provided by smithy.
+  /// This named constructor should be used internally only.
```

Comment:
```suggestion
  /// Creates a [StorageS3Item] from [S3Object] provided by smithy.
  ///
  /// This named constructor should be used internally only.
```

https://dart.dev/guides/language/effective-dart/documentation#do-separate-the-first-sentence-of-a-doc-comment-into-its-own-paragraph